app/models/reservation_scopes.rb

```diff
@@ -1,51 +1,37 @@
   def self.included(base) # rubocop:disable MethodLength, AbcSize
     base.class_eval do
-      scope :recent, ->() { order('start_date, due_date, reserver_id') }
+      default_scope { order('start_date, due_date, reserver_id') }
       scope :user_sort, ->() { order('reserver_id') }
-      scope :finalized, lambda {
-        where('approval_status = ? OR approval_status = ?', 'auto',
-              'approved')
-      }
-      scope :not_returned, ->() { where('checked_in IS NULL').finalized }
-      scope :active, ->() { not_returned }
-      # TODO: grep codebase for active and replace with not_returned or
-      # figure out a better way to give the same scope two names
-      scope :untouched, ->() { where('checked_out IS NULL').not_returned }
-      scope :reserved, lambda {
-        where('due_date >= ?', Time.zone.today).untouched.recent
-      }
-      scope :checked_out, lambda {
-        where('checked_out IS NOT NULL').not_returned.recent
+
+      scope :not_available, lambda {
+        where(status: Reservation.statuses.values_at(
+          *%w(reserved checked_out)).compact)
       }
+      scope :active, ->() { not_available }
+      scope :finalized, ->() { not_available }
+
       scope :checked_out_today, lambda {
         where('checked_out >= ? and checked_out <= ?',
               Time.zone.today,
-              Time.zone.today + 1.day).not_returned.recent
+              Time.zone.today + 1.day).checked_out
       }
       scope :checked_out_previous, lambda {
         where('checked_out < ? and due_date <= ?', Time.zone.today,
-              Time.zone.today + 1.day).not_returned.recent
-      }
-      scope :overdue, lambda {
-        where('due_date < ?', Time.zone.today).checked_out
-      }
-      scope :returned, lambda {
-        where('checked_in IS NOT NULL and checked_out IS NOT NULL').recent
+              Time.zone.today + 1.day).checked_out
       }
+      scope :overdue, ->() { where(overdue: true).checked_out }
+
       scope :checked_in, ->() { returned }
 
-      # TODO: the following two scopes are broken since the due date falls
-      # within the range of allowable return dates (since it's a Date). This
-      # will be fixed by using a :status column that will be changed over time
-      # (see #462)
-      scope :returned_on_time, ->() { where('checked_in <= due_date').returned }
-      scope :returned_overdue, ->() { where('due_date < checked_in').returned }
-      scope :missed, lambda {
-        where('due_date < ?', Time.zone.today).untouched.recent
-      }
+      scope :returned_on_time, ->() { where(overdue: false).returned }
+      scope :returned_overdue, ->() { where(overdue: true).returned }
       scope :upcoming, lambda {
-        where('start_date = ?', Time.zone.today).reserved.user_sort
+        unscoped.where(start_date: Time.zone.today).reserved.user_sort
+      }
+      scope :due_soon, lambda {
+        where('due_date >= ? and due_date < ?', Time.zone.today,
```

Comment:
Since the `due_date` field is stored as a `Date`, this should basically be equivalent to `where(due_date: Time.zone.today).checked_out`, unless I'm missing something... I haven't gone through everything else yet to see where this is used, but it seems like those two queries would be equivalent.